sdk/core/core-client-rest/test/urlHelpers.spec.ts

```diff
@@ -168,4 +168,72 @@ describe("urlHelpers", () => {
     });
     assert.equal(result, "https://foo%bar.org");
   });
+
+  it("allows for path parameters with metadata", () => {
+    const result = buildRequestUrl("https://example.org/", "{foo}/", [
+      {
+        value: "foo",
+      },
+    ]);
+
+    assert.equal(result, "https://example.org/foo/");
+  });
+
+  it("allowReserved in path parameter defaults to false", () => {
+    const result = buildRequestUrl("https://example.org/", "{foo}/", [
+      {
+        value: "foo/bar",
+      },
+    ]);
+
+    assert.equal(result, "https://example.org/foo%2Fbar/");
+  });
+
+  it("allowReserved in path parameter allows special characters in path parameters", () => {
+    const result = buildRequestUrl("https://example.org/", "{foo}/", [
+      {
+        value: "foo/bar",
+        allowReserved: true,
+      },
+    ]);
+
+    assert.equal(result, "https://example.org/foo/bar/");
+  });
+
+  it("allows for query parameters with metadata", () => {
+    const result = buildRequestUrl("https://example.org/", "", [], {
+      queryParameters: {
+        foo: {
+          value: "bar",
+        },
+      },
+    });
+
+    assert.equal(result, "https://example.org/?foo=bar");
+  });
+
+  it("explode defaults to false", () => {
+    const result = buildRequestUrl("https://example.org/", "", [], {
+      queryParameters: {
+        foo: {
+          value: ["bar", "baz"],
+        },
+      },
+    });
+
+    assert.equal(result, "https://example.org/?foo=bar,baz");
+  });
+
+  it("explode decomposes query parameter array into multiple query parameters", () => {
```

Comment:
do we have a case for array directly? as we confirmed offline this would be a non-breaking in core?

```
const result = buildRequestUrl("https://example.org/", "", [], {
      queryParameters: {
        foo:  ["bar", "baz"]
      },
    });
```

Reply:
we have [this one](https://github.com/Azure/azure-sdk-for-js/pull/31058/files#diff-e4db553b388bf7e6e909a8cd55fed3b41bf1989363c00e532e90d1e782c4fa5aR215-R225), is that what you're looking for?

Reply:
yes and thanks for pointing out.